### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by achilehero** [bithost-gmbh/ngx-mat-select-search#298]
**Describe the bug**
Inconsistent behaviour of the toggleAll checkbox when handling selected options

**To Reproduce**
1. Use the `Multiple selection with Select All Checkbox` example.
2. Open the dropdown and clear the selection
3. Type `swi` in the search to filter some of the results. That way we have two possible options displayed
4. Click the togleAll checkbox to select all the options. toggleAll state is displayed as indeterminate (correct)
5. Deselect both options by clicking once on each of them. Now the toggleAll checkbox state is set to checked (wrong. it should be unchecked). See the image.

*Important* Please provide a reproduction example. You can use https://stackblitz.com/github/bithost-gmbh/ngx-mat-select-search-example as a starting point.
The above mentioned example can be used.

**Expected behavior**
Behavior is described in reproduction steps: toggleAll checkbox state should be set according to the individual selections.

**Screenshots**
![image](https://user-images.githubusercontent.com/14956052/104724745-c54f6800-5739-11eb-9ab2-22f099b54857.png)

**Desktop (please complete the following information):**

 - OS: Windows 10
 - Browser: Chrome
 - Version 87.0.4280.88

**Comment by macjohnny** [comment]
this problem occurs because 
https://github.com/bithost-gmbh/ngx-mat-select-search/blob/38a93b3c5911367bf7d002849cfac69be948b638/src/app/examples/06-multiple-selection-select-all-example/multiple-selection-select-all-example.component.html#L9
evaluates to `false` in all steps 1. to 5.

- after step 3: the checkbox is in the state
  - checked: false
  - indeterminate: false
- after step 4.: by clicking the select all checkbox ., the checkbox is in the checked state
  - checked: true
  - indeterminate: true
- after step 5: by removing the selection, the checkbox is not in the indeterminate state anymore, however, the checked state is unchanged
  - checked: true
  - indeterminate: false

the reason why the checked state does not change is that the input value here does not change after initialization (i.e. before step 1)

one way to resolve this would be to toggle the selection state of all options when clicking the checkbox, not just the existing ones.

**Comment by achilehero** [comment]
True. And another way to resolve it would be (like I had the case in my app) to compare it with the filtered selection (i.e. have `(filteredBanksMulti|async).length` instead of `allBanksSize`. Do you you want me to push a PR with that so you could check it?

**Comment by macjohnny** [comment]
@achilehero a PR would be appreciated

**Comment by achilehero** [comment]
https://github.com/bithost-gmbh/ngx-mat-select-search/pull/300
